components/brave_wallet/renderer/js_cardano_provider.cc

```diff
@@ -237,18 +232,12 @@ void JSCardanoProvider::Install(content::RenderFrame* render_frame) {
                            gin::StringToV8(isolate, kCardano), true);
   }
 
-  gin::Handle<JSCardanoProvider> cardano_brave_provider =
-      gin::CreateHandle(isolate, new JSCardanoProvider(render_frame));
-  if (cardano_brave_provider.IsEmpty()) {
-    return;
-  }
-  v8::Local<v8::Value> cardano_brave_provider_value =
-      cardano_brave_provider.ToV8();
-  v8::Local<v8::Object> cardano_brave_provider_object;
-  if (!cardano_brave_provider_value->ToObject(context).ToLocal(
-          &cardano_brave_provider_object)) {
-    return;
-  }
+  JSCardanoProvider* cardano_brave_provider =
+      cppgc::MakeGarbageCollected<JSCardanoProvider>(
+          isolate->GetCppHeap()->GetAllocationHandle(), render_frame);
+
+  v8::Local<v8::Object> cardano_brave_provider_object =
+      cardano_brave_provider->GetWrapper(isolate).ToLocalChecked();
```

Comment:
Same question if ToLocalChecked is safe to call like that

Reply:
cc @cypt4